File: 70-climbing-stairs/70-climbing-stairs.cpp
unordered_map<int , int> memo;
unordered_map<int,int>::iterator it;
int stair(int n)
{
	it = memo.find(n);
	if (it != memo.end())
	{
		return memo[n];
	}
	if(n<2){
		return 1;
	}
	memo[n] = stair(n-1) + stair(n-2);
	return memo[n];
}
class Solution {
public:
    int climbStairs(int n) {
       return stair(n);
    }
};

File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    def hell(self, n: int , memo) -> int:
        if n in memo:
            return memo[n]
        if n==1:
            return 1
        if n==2:
            return 2
        memo[n] = self.hell(n-1 , memo) + self.hell(n-2 , memo)
        return memo[n]
    def climbStairs(self, n: int) -> int:
        memo = {}
        memo[1] = 1
        memo[2] = 2
        return self.hell(n , memo)    